python/cuml/dask/datasets/make_regression.py

```diff
@@ -0,0 +1,216 @@
+        Determines random number generation for dataset creation. Pass an int
+        for reproducible output across multiple function calls.
+    n_parts : int, optional (default=1)
+        The number of parts of work.
+
+    Returns
+    -------
+    X : Dask-cuPy array of shape [n_samples, n_features]
+        The input samples.
+    y : Dask-cuPy array of shape [n_samples] or [n_samples, n_targets]
+        The output values.
+    coef : Dask-cuPy array of shape [n_features]
+           or [n_features, n_targets], optional
+        The coefficient of the underlying linear model. It is returned only if
+        coef is True.
+    """
+    n_informative = min(n_features, n_informative)
+    rs = da.random.RandomState(seed=random_state,
+                               RandomState=cp.random.RandomState)
+
+    n_samples_per_part = max(1, int(n_samples / n_parts))
+
+    if effective_rank is None:
+        # Randomly generate a well conditioned input set
+        X = rs.standard_normal((n_samples, n_features),
+                               chunks=(n_samples_per_part, (n_informative,
+                                                            n_features -
+                                                            n_informative)))
+
+    else:
+        # Randomly generate a low rank, fat tail input set
+        X = make_low_rank_matrix(n_samples=n_samples,
+                                 n_features=n_features,
+                                 effective_rank=effective_rank,
+                                 tail_strength=tail_strength,
+                                 random_state=random_state,
+                                 n_parts=n_parts)
+        X = X.rechunk({0: n_samples_per_part,
+                       1: (n_informative, n_features-n_informative)})
+
+    # Generate a ground truth model with only n_informative features being non
+    # zeros (the other features are not correlated to y and should be ignored
+    # by a sparsifying regularizers such as L1 or elastic net)
+
+    ground_truth = 100.0 * rs.standard_normal((n_informative, n_targets),
+                                              chunks=(n_samples_per_part, -1))
+    y = da.dot(X[:, :n_informative], ground_truth) + bias
+
+    if n_informative != n_features:
+        zeroes = 0.0 * rs.standard_normal((n_features -
+                                           n_informative,
+                                           n_targets))
+        ground_truth = da.concatenate([ground_truth, zeroes], axis=0)
+
+    # Add noise
+    if noise > 0.0:
+        y += rs.normal(scale=noise, size=y.shape)
+
+    # Randomly permute samples and features
+    if shuffle:
+        samples_indices = rs.permutation(n_samples)
+        X = X[samples_indices, :]
+        y = y[samples_indices, :]
+
+        features_indices = rs.permutation(n_features)
+        X = X[:, features_indices]
+        ground_truth = ground_truth[features_indices, :]
+
+    y = cp.squeeze(y)
```

Comment:
Use `rmm_cupy_ary` as mentioned above

Reply:
Actually y is a Dask-CuPy array here. So, I think `y = da.squeeze(y)` will do the job. Unfortunately, I do not know if the random initialization of the Dask-CuPy arrays are able to be tracked by the RMM.

Reply:
`da.squeeze` is perfect and more immediately obvious.